routes/schema.js

```diff
@@ -0,0 +1,77 @@
+
+const { Joi } = require('celebrate');
+
+function _depositAsset({ TOKEN_ADDRESS_WHITELIST }){
+    let schema = Joi.string().required()
+    if( TOKEN_ADDRESS_WHITELIST && TOKEN_ADDRESS_WHITELIST.length > 0 ){
+        schema = schema.valid(TOKEN_ADDRESS_WHITELIST)
+    }
+    return schema
+}
+
+function _depositAmount({ MINIMAL_DEPOSIT  }){
+    let schema = Joi.number().required()
+    if( MINIMAL_DEPOSIT > 0 ){
+        schema = schema.min(MINIMAL_DEPOSIT)
+    }
+    return schema
+}
+
+function _creator({ CREATORS_WHITELIST }){
+    let schema = Joi.array().items(Joi.string()).required().length(2)
+    if(CREATORS_WHITELIST && CREATORS_WHITELIST.length > 0 ){
+        schema = schema.has(CREATORS_WHITELIST)
+    }
+    return schema
+
+}
+
+function _validators({ CREATORS_WHITELIST }) {
+    let schema = Joi.array().items(
+        Joi.object({
+            "id": Joi.string().required(),
+            "url": Joi.string().uri({
+                scheme: ['http', 'https']
+            }).required(),
+            "fee": Joi.number().required()
+        })
+    )
+    
+    if(CREATORS_WHITELIST && CREATORS_WHITELIST.length > 0 ){
+        schema = schema.has(Joi.object({
+            "id": Joi.any().valid(CREATORS_WHITELIST),
+            "url": Joi.string().uri({
+                scheme: ['http', 'https']
+            }).required(),
+            "fee": Joi.number().required()
+        }))
+    }
+
+    return schema.length(2)
+}
+module.exports = {
+    createCampaign: (cfg) => ({
+        "id": Joi.string().required(),
+        "depositAsset":  _depositAsset(cfg),
+        "depositAmount": _depositAmount(cfg),
+        "validators":   _creator(cfg), 
+        "spec" : Joi.object({ 
+            "validators" : _validators(cfg)
+        }),
+        "watcher": Joi.object({
+            "ethereum": Joi.object({
+                "contract": Joi.string().required()
+            })
+        })
+    }),
+    validateCampaign: (cfg) => ({
+        "id" : Joi.string().required(),
```

Comment:
no quottion marks again; maybe we can configure jslint to enforce this

Reply:
done